tests/unit/customizations/s3/test_s3handler.py

```diff
@@ -903,190 +906,152 @@ def test_remove_bucket(self):
-                s3handler._enqueue_tasks([fileinfo])
-                # Assert that the size of the ``FileInfo`` object was set
-                # if we are downloading a stream.
-                self.assertTrue(mock_set_size_from_s3.called)
-                # Ensure that this download would have been a multipart
-                # download.
-                self.assertTrue(mock_enqueue_range_tasks.called)
-
-    def test_enqueue_range_download_tasks_stream(self):
-        s3handler = S3StreamHandler(
-            self.session, self.params,
-            runtime_config=runtime_config(multipart_chunksize=100))
-        s3handler.executor = mock.Mock()
-        fileinfo = FileInfo('filename', operation_name='download',
-                            is_stream=True, size=100)
-        s3handler._enqueue_range_download_tasks(fileinfo)
-        # Ensure that no request was sent to make a file locally.
-        submitted_tasks = s3handler.executor.submit.call_args_list
-        self.assertNotEqual(type(submitted_tasks[0][0][0]),
-                            CreateLocalFileTask)
+        if maximum is None:
+            maximum = awscli.customizations.s3.utils.MAX_SINGLE_UPLOAD_SIZE
+        if minimum is None:
+            minimum = awscli.customizations.s3.utils.MIN_UPLOAD_CHUNKSIZE
+
+        self.assertLessEqual(size, maximum)
+        self.assertGreaterEqual(size, minimum)
+
+    def test_upload_stream(self):
+        handler = S3TransferStreamHandler(self.session, self.params)
+        file = FileInfo('-', 'foo-bucket/bar.txt', is_stream=True,
+                        operation_name='upload')
+
+        with MockStdIn(b'foobar'):
+            response = handler.call([file], self.transfer_manager)
+
+        self.assertEqual(response.num_tasks_failed, 0)
+        self.assertEqual(response.num_tasks_warned, 0)
+
+        upload_args = self.transfer_manager.upload.call_args[1]
+        self.assertEqual(upload_args['bucket'], 'foo-bucket')
+        self.assertEqual(upload_args['key'], 'bar.txt')
+
+    def test_upload_stream_with_expected_size(self):
+        expected_size = 6
+        self.params['expected_size'] = expected_size
+        handler = S3TransferStreamHandler(self.session, self.params)
+        file = FileInfo('-', 'foo-bucket/bar.txt', is_stream=True,
+                        operation_name='upload')
+
+        with MockStdIn(b'foobar'):
+            handler.call([file], self.transfer_manager)
+
+        # Assert that there is a subscriber.
+        call_args = self.transfer_manager.upload.call_args[1]
+        subscribers = call_args.get('subscribers', [])
+        self.assertTrue(len(subscribers) == 1)
+
+        # Make sure that subscriber is the right kind
+        subscriber = subscribers[0]
+        self.assertIsInstance(subscriber, ProvideSizeSubscriber)
+
+        # Validate that the size on the subscriber is the expected size
+        self.assertEqual(subscriber.size, expected_size)
+
```

Comment:
True. How hard would a functional or integration test be for this? I feel it may be good to have a quick test such that we make sure we have everything wired up correctly.